tools/link.py

```diff
@@ -1127,11 +1127,13 @@ def limit_incoming_module_api():
         settings.DEFAULT_LIBRARY_FUNCS_TO_INCLUDE += ['$warnOnce']
 
       settings.DEFAULT_LIBRARY_FUNCS_TO_INCLUDE += ['$getValue', '$setValue']
-      # TODO(sbc): Remove these forced dependencies.
-      settings.DEFAULT_LIBRARY_FUNCS_TO_INCLUDE += ['$runDependencies', '$addRunDependency', '$removeRunDependency']
 
     settings.DEFAULT_LIBRARY_FUNCS_TO_INCLUDE += ['$ExitStatus']
 
+  # Certain configurations require the removeRunDependency/addRunDependency system.
+  if settings.LOAD_SOURCE_MAP or (settings.WASM_ASYNC_COMPILATION and not settings.MODULARIZE):
+    settings.DEFAULT_LIBRARY_FUNCS_TO_INCLUDE += ['$runDependencies', '$addRunDependency', '$removeRunDependency']
```

Comment:
What about `tools/file_packager.py`? Some of the add/removes there are covered by the `src/lib/libfs_shared.js` change in this PR, but that just handles file preloading, and there is a bunch of other stuff there?

Reply:
Good catch.  Added for that case too.

Reply:
In file_packager.py I also see LZ4FS uses it, am I reading that wrong?